### PR TITLE
feat(provider/kubernetes): v2 cache & surface events per-resource

### PR DESCRIPTION
**Pull request by lwander**
https://github.com/spinnaker/spinnaker/issues/2587

### CODE REVIEW

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/view/model/KubernetesV2Manifest.java [hunk 1]
```diff
@@ -40,4 +40,5 @@
   private KubernetesManifest manifest;
   private Status status;
   private Set<Artifact> artifacts = new HashSet<>();
+  private Set<KubernetesManifest> events = new HashSet<>();
```
Comment (wjoel) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/view/model/KubernetesV2Manifest.java]: I would have expected this to be a list of `V1Event`. Shouldn't it return the events shown in `kubectl describe <kind> <name>`, similar to `kubectl get events --field-selector involvedObject.name=name-of-my-pod,involvedObject.kind=Pod`? I think that's what we'd like to show in the UI.

Comment (lwander) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/view/model/KubernetesV2Manifest.java]: `V1Event` can be mapped to and from `KubernetesManifest`. I'm afraid of embedding those models in here -- we have no assurance that those will work across versions of future kubernetes clusters (what if `V2Event` gets introduced for all new resources starting at 1.10?)

Comment (lwander) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/view/model/KubernetesV2Manifest.java]: Just to be clear, it is exactly storing the events you're pointing out in your sample `kubectl` commands.

Comment (wjoel) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/view/model/KubernetesV2Manifest.java]: I see. It's a bit surprising (those are far from the kinds of manifests I'm used to) but it makes sense to keep the interface at that level. Just curious, then: why not a (sorted) list? We'll want to use the events sorted by time pretty much always, might as well generate a sorted list here?

Comment (lwander) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/view/model/KubernetesV2Manifest.java]: Yup all resources follow that pattern.

Good point, I'll sort by `lastTimestamp`.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/view/provider/KubernetesV2ManifestProvider.java [hunk 2]
```diff
@@ -73,6 +78,14 @@ public KubernetesV2Manifest getManifest(String account, String location, String
       return null;
     }
 
+    Function<KubernetesManifest, String> lastEventTimestamp = (m) -> (String) m.get("lastTimestamp");
```
Comment (wjoel) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/view/provider/KubernetesV2ManifestProvider.java]: https://kubernetes.io/docs/reference/generated/federation/v1/definitions/#_v1_event lists this field as not required, so we should probably have a reasonable default.

Comment (lwander) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/view/provider/KubernetesV2ManifestProvider.java]: Good point - I'll fall back to the "firstTimestamp", and then to "N/A" which'll sort higher than any timestamp

**Pull request merged**